nntrainer/src/input_layer.cpp

```diff
@@ -90,23 +92,26 @@ Tensor InputLayer::forwarding(Tensor in, int &status) {
 
 int InputLayer::initialize(bool last) {
   int status = ML_ERROR_NONE;
-  if (dim.batch() <= 0 || dim.channel() <= 0 || dim.height() <= 0 ||
-      dim.width() <= 0) {
+  if (input_dim.batch() <= 0 || input_dim.channel() <= 0 ||
+      input_dim.height() <= 0 || input_dim.width() <= 0) {
     ml_loge("Error: Dimension must be greater than 0");
     return ML_ERROR_INVALID_PARAMETER;
   }
 
+  dim = input_dim;
```

Comment:
Q. Is `dim` still in use somewhere? I think it is better to be delete `dim` or change it's name because semantically it could mean both `input dim` and `output dim` or else.

Reply:
no. it is going to be used for conv2d.